wallet/src/main/scala/org/bitcoins/wallet/api/CoinSelector.scala

```diff
@@ -0,0 +1,69 @@
+package org.bitcoins.wallet.api
+
+import org.bitcoins.core.currency.{CurrencyUnit, CurrencyUnits}
+import org.bitcoins.core.protocol.transaction.TransactionOutput
+import org.bitcoins.core.wallet.fee.FeeUnit
+import org.bitcoins.wallet.models.UTXOSpendingInfoDb
+
+import scala.annotation.tailrec
+
+/** Implements algorithms for selecting from a UTXO set to spend to an output set at a given fee rate. */
+trait CoinSelector {
+
+  /** Greedily selects from walletUtxos starting with the largest outputs, skipping detrimental outputs */
+  def accumulateLargest(
+      walletUtxos: Seq[UTXOSpendingInfoDb],
+      outputs: Seq[TransactionOutput],
+      feeRate: FeeUnit): Vector[UTXOSpendingInfoDb] = {
+    val sortedUtxos =
+      walletUtxos.toVector.sortBy(_.value.satoshis.toLong).reverse
+
+    accumulate(sortedUtxos, outputs, feeRate)
+  }
+
+  /** Greedily selects from walletUtxos starting with the smallest outputs, skipping detrimental outputs */
```

Comment:
So I think the inverse of the comment I have above applies here. This scheme might be best used in low fee environments, and another tradeoff with this coin selection is that you are reducing your privacy by linking `n` utxos into a single transaction. Not that that is necessarily a bad thing -- it has to be done eventually -- but it is good to note the trade off.